engine/lib/print_rust.ml

```diff
@@ -278,8 +278,14 @@ module Raw = struct
     | Match { scrutinee; arms } ->
         let arms =
           List.map
-            ~f:(fun { arm = { arm_pat; body }; _ } ->
-              ppat arm_pat & !" => {" & pexpr body & !"}")
+            ~f:(fun { arm = { arm_pat; body; guard }; _ } ->
+              let g =
+                match guard with
+                | Some { guard = IfLet { lhs; rhs; _ }; _ } ->
+                    !" if let " & ppat lhs & !" = " & pexpr rhs
+                | None -> !""
+              in
```

Comment:
Similar comments that on the generic printer: maybe use `Option.map`&`Option.value` and name `g` `guard`?